### PR TITLE
feat: basic multichain config

### PR DESCRIPTION
**Pull request by cnasc**
**What changed? Why?**
Adding some types and a config file that will make it easier for us to support multiple chains. Pushing this up before any further implementation to get feedback on the approach taken here.

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

**Comment by cnasc** [comment]
@robpolak how does this look?

File: apps/build-onchain-apps/src/contract/ContractSpecification.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Abi, type Chain } from 'viem';
+import { baseGoerli } from 'viem/chains';
+import BuyMeACoffeeABI from './BuyMeACoffee';
+
+export type ContractInstance = {
+  chain: Chain;
+  address: `0x${string}`,
```
Comment (robpolak) [inline apps/build-onchain-apps/src/contract/ContractSpecification.ts:7]: What is this black magic

**Review by robpolak — APPROVED** [review]
LGTM

File: apps/build-onchain-apps/src/contract/ContractSpecification.ts (hunk 1 above)
Comment (cnasc) [reply apps/build-onchain-apps/src/contract/ContractSpecification.ts:7]: Basically “a string that starts with `”0x”`. Compiler can only tell from string literals naturally

Comment (Zizzamia) [reply apps/build-onchain-apps/src/contract/ContractSpecification.ts:7]: Harry Potter vibes

File: apps/build-onchain-apps/src/contract/ContractSpecification.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Abi, type Chain } from 'viem';
+import { baseGoerli } from 'viem/chains';
+import BuyMeACoffeeABI from './BuyMeACoffee';
+
+export type ContractInstance = {
```
Comment (Zizzamia) [inline apps/build-onchain-apps/src/contract/ContractSpecification.ts:5]: if we are exporting `types`, should at this point have just have a `types.ts` file in this folder that we use to organize different kind of Contract types we need?

Comment (cnasc) [reply apps/build-onchain-apps/src/contract/ContractSpecification.ts:5]: I usually prefer to have the types colocated with any implementation code, but open to your approach as well. Do you have a strong preference?

**Review by Zizzamia — APPROVED** [review]

File: apps/build-onchain-apps/src/contract/ContractSpecification.ts (hunk 2 above)
Comment (Zizzamia) [reply apps/build-onchain-apps/src/contract/ContractSpecification.ts:5]: Approved and merged the PR as this is not a blocker.

My thinking is about where and how things can be re-use. I don't want to jump on conclusion, so happy to just revisit this as we have more code to read :-)

LFG

**Pull request merged**